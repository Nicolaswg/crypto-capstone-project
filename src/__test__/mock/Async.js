import React, { useEffect, useState } from 'react';
import axios from 'axios';

const Async = () => {
  const [coin, setCoin] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const response = await axios.get('https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1');
      setCoin(response);
    };
    fetchData();
  }, []);

  return (
    <div>
      <ul data-testid="list">
        {coin.map((item) => (
          <li key={item.id}>{item.name}</li>
        ))}
      </ul>
    </div>
  );
};

export default Async;
