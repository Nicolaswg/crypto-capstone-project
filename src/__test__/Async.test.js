import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import axios from 'axios';
import Async from './mock/Async';

jest.mock('axios');

afterEach(cleanup);

it('renders Async correctly', async () => {
  axios.get.mockResolvedValue([
    { id: 'bitcoin', symbol: 'btc', name: 'Bitcoin' },
    { id: 'ethereum', symbol: 'eth', name: 'Ethereum' },
  ]);
  const { getByTestId, asFragment } = render(<Async />);

  const listNode = await waitFor(() => getByTestId('list'));
  expect(listNode.children).toHaveLength(2);
  expect(asFragment()).toMatchSnapshot();
});
