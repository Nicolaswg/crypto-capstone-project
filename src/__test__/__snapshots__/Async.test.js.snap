// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Async correctly 1`] = `
<DocumentFragment>
  <div>
    <ul
      data-testid="list"
    >
      <li>
        Bitcoin
      </li>
      <li>
        Ethereum
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
